### PR TITLE
🐛Source Outreach: Fix HTTP 404 during cursor pagination

### LINKED ISSUE #41107: [source-outreach] HTTP 404 during pagination
### Connector Name

source-outreach

### Connector Version

1.0.4

### What step the error happened?

During the sync

### Relevant information

Any stream that doesn't require pagination works perfectly. When a stream does require pagination, it gets the first 1000 records fine. When it goes to get the next 1000 records, the connector gets an HTTP 404 error.

It looks like the paginator is sending requests to `api.outreach.io/<stream>` instead of sending them to `api.outreach.io/api/v2/<stream>`. It also looks like it's using Python 3.9. That's a little surprising but I don't know whether it matters. Those are the only things I noticed in the logs.

Here are Outreach's docs about how their pagination works:
https://developers.outreach.io/api/making-requests/#cursor-based-pagination

airbyte version: 0.63.5
source-outreach version: 1.0.4

Airbyte is self-hosted on K8s and deployed using Airbyte's Helm chart.

### Relevant log output

```shell
2024-07-09 18:14:05 platform > Executing worker wrapper. Airbyte version: 0.63.5

[...]

2024-07-09 18:14:05 platform > Attempting to start pod = source-outreach-check-10552-0-vsmqm for airbyte/source-outreach:1.0.4

[...]

2024-07-09 18:15:47 source > Encountered an exception while reading stream accounts
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 135, in read
    yield from self._read_stream(
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 230, in _read_stream
    for record_data_or_message in record_iterator:
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/streams/core.py", line 169, in read
    for record_data_or_message in records:
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/declarative_stream.py", line 126, in read_records
    yield from self.retriever.read_records(self.get_json_schema(), stream_slice)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py", line 327, in read_records
    for stream_data in self._read_pages(record_generator, self.state, _slice):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py", line 291, in _read_pages
    response = self._fetch_next_page(stream_state, stream_slice, next_page_token)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py", line 266, in _fetch_next_page
    return self.requester.send_request(
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/requesters/http_requester.py", line 461, in send_request
    return self._validate_response(response)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/requesters/http_requester.py", line 566, in _validate_response
    raise ReadException(error_message)
airbyte_cdk.sources.declarative.exceptions.ReadException: Request to https://api.outreach.io/accounts?count=false&page%5Bafter%5D=eyJjbiI6IkFwaVYyOjpBY2NvdW50c0NvbnRyb2xsZXIiLCJzcCI6InVwZGF0ZWRBdCIsInN2IjpbIjIwMjQtMDUtMjBUMTk6NDM6MDUuMDAwWiJdLCJpZCI6OTUxOCwidiI6MX0%3D&page%5Bsize%5D=1000&sort=updatedAt failed with status code 404 and error message None

[...]

2024-07-09 18:15:47 source > During the sync, the following streams did not sync successfully: accounts: AirbyteTracedException('Request to https://api.outreach.io/accounts?count=false&page%5Bafter%5D=eyJjbiI6IkFwaVYyOjpBY2NvdW50c0NvbnRyb2xsZXIiLCJzcCI6InVwZGF0ZWRBdCIsInN2IjpbIjIwMjQtMDUtMjBUMTk6NDM6MDUuMDAwWiJdLCJpZCI6OTUxOCwidiI6MX0%3D&page%5Bsize%5D=1000&sort=updatedAt failed with status code 404 and error message None')
2024-07-09 18:15:47 source > None
Traceback (most recent call last):
  File "/airbyte/integration_code/main.py", line 8, in <module>
    run()
  File "/airbyte/integration_code/source_outreach/run.py", line 14, in run
    launch(source, sys.argv[1:])
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 235, in launch
    for message in source_entrypoint.run(parsed_args):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 122, in run
    yield from map(AirbyteEntrypoint.airbyte_message_to_string, self.read(source_spec, config, config_catalog, state))
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 166, in read
    for message in self.source.read(self.logger, config, catalog, state):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/declarative/manifest_declarative_source.py", line 167, in read
    yield from super().read(logger, config, catalog, state)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 184, in read
    raise AirbyteTracedException(message=error_message, failure_type=FailureType.config_error)
airbyte_cdk.utils.traced_exception.AirbyteTracedException: None
```

### Contribute

- [ ] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
Closes https://github.com/airbytehq/airbyte/issues/41107

## How
Fixed the base URL to include closure for identifying the parsing point

## User Impact
Bug fix

### CODE REVIEW

**Comment by btkcodedev** [comment]
The issue was with base URL without closure ("/"), the declarative default paginator initially sets the cursor to the base URL
Thus if the parser can't find the closure, it resets to initial parsed token with other parameters
Tested & Verified the fix with builder: :white_check_mark: 
image

Tested with terminal: :white_check_mark: 
image

**Comment by btkcodedev** [comment]
Ready for review and merge

**Comment by btkcodedev** [comment]
Ah, yes, just a tiny, super simple fix—only took an eternity to find the time. 🤷‍♂️

File: airbyte-integrations/connectors/source-outreach/source_outreach/manifest.yaml [hunk 1]
```diff
@@ -180,7 +180,7 @@ definitions:
 
   requester:
     type: HttpRequester
-    url_base: https://api.outreach.io/api/v2
+    url_base: https://api.outreach.io/api/v2/
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-outreach/source_outreach/manifest.yaml:183]: Slightly weird — were you able to test this, @btkcodedev? I assumed that the CDK would attempt and add / in automaticallly. But let's see!

**Review by natikgadzhi — APPROVED** [review]

**Comment by btkcodedev** [comment]
Yes, I've tested it with both CLI and builder, it was the reason. 🎊

File: airbyte-integrations/connectors/source-outreach/source_outreach/manifest.yaml (hunk 1 above)
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-outreach/source_outreach/manifest.yaml:183]: Yes @natikgadzhi, I've tested it with both CLI and builder, it was the reason. 🎊

**Pull request merged**